.golangci.yml

```diff
@@ -22,19 +22,30 @@ linters-settings:
     # Disable SA1019 to allow use of deprecated label
     checks: ["all", "-SA1019"]
 
+  custom:
+    structwrite:
+      type: module
+      description: "disallow struct field writes outside constructor"
+      original-url: "github.com/onflow/flow-go/tools/structwrite"
+      settings:
+        structs:
+          - "github.com/onflow/flow-go/model/flow.Header"
```

Comment:
The way it is currently set up, we opt into the linter by adding fully qualified type names here. Alternatively we could add a recognizable line to the godoc of the relevant struct definition, like:
```go
//structwrite: This type must not be modified outside the constructor
type Header struct {
// ...
```

Reply:
I personally prefer the linter rule to be right were the struct definition can be found. Maybe we can allow both ways of informing the linter that a struct shouldn't be written to?